docs/datamodel/functions.rst

```diff
@@ -29,6 +29,34 @@ cardinality:
     {2}
 
 
+User-defined Functions
+----------------------
+
+It is also possible to define custom functions. For example, consider
+a function that adds an exclamation mark ``'!'`` at the end of the
+string or produces ``'!!!'`` if the input is omitted or empty:
+
+.. code-block:: sdl
+
+    function exclamation(word: optional str = {}) -> str
+        using ((word ++ '!') ?? '!!!');
+
+In order to make sure that the function is called when the argument is
+an empty set ``{}`` we make the argument :ref:`optional
+<ref_eql_fundamentals_optional>`. We also provide a default value of
+``{}`` if the argument is omitted entirely. Here are some results this
+function produces:
```

Comment:
Yeah, the "optional" stuff is cool, but the whole purpose of this section is to say just one thing: "you can define your own functions". That's it. This section isn't about anything else, and especially it's not about `optional` arguments.

Reply:
FIxed.